File: 101-200/155.Min Stack.cpp
class MinStack {
private:
    stack<int> s1;
    stack<int> s2;
public:
    /** initialize your data structure here. */
    MinStack() {
        
    }
    
    void push(int x) {
        s1.push();
        if (s2.empty() || x<getMin()){
            s2.push();
        }
    }
    
    void pop() {
        if(s1.top()==getMin()){
            s2.pop();
        }
        s1.pop()
    }
    
    int top() {
        return s1.top();
    }
    
    int getMin() {
        return s2.top();
    }
};

/**
 * Your MinStack object will be instantiated and called as such:
 * MinStack obj = new MinStack();
 * obj.push(x);
 * obj.pop();
 * int param_3 = obj.top();
 * int param_4 = obj.getMin();
 */

File: 101-200/155.Min Stack.py
class MinStack(object):

    def __init__(self):
        self.a=[]
        

    def push(self, x):
        nowmin=self.getMin()
        if nowmin==None or nowmin>x:
            nowmin=x
        self.a.append((x,nowmin))

    def pop(self):
        self.a.pop()
        

    def top(self):
        if len(self.a)==None:
            return None
        else:
            return self.a[len(self.a)-1][0]

        

    def getMin(self):
        if len(self.a)==None:
            return None 
        else:
            return self.a[len(self.a)-1][1]
        


# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(x)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()